packages/typegpu/src/data/matrix.ts

```diff
@@ -123,11 +92,6 @@ abstract class mat2x2Impl<TColumn extends vec2f> implements mat2x2<TColumn> {
 
   abstract makeColumn(e0: number, e1: number): TColumn;
 
-  *elements() {
-    yield* this.columns[0];
-    yield* this.columns[1];
-  }
```

Comment:
I moved functionality to operate ON data, instead of being part of the data. This allows for native matrices and exotic matrices to match on the type level.

So something like the following:
```ts
const array = [...viewMatrix];
```
Will have to be written like this:
```ts
const array = mat3x3fToArray(viewMatrix);
```
We could compromise and make one function that turns any matrix to an array:
```ts
const array = matToArray(viewMatrix);
```